server/channels/app/post_persistent_notification.go

```diff
@@ -0,0 +1,367 @@
+			// Either the notification post is already deleted or was never a notification post
+			return nil
+		default:
+			return model.NewAppError("DeletePersistentNotificationsPost", "app.post_priority.delete_persistent_notification_post.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+	}
+
+	if !*a.Config().ServiceSettings.AllowPersistentNotificationsForGuests {
+		user, nErr := a.Srv().Store().User().Get(context.Background(), loggedInUserID)
+		if nErr != nil {
+			var nfErr *store.ErrNotFound
+			switch {
+			case errors.As(nErr, &nfErr):
+				return model.NewAppError("DeletePersistentNotificationsPost", MissingAccountError, nil, "", http.StatusNotFound).Wrap(nErr)
+			default:
+				return model.NewAppError("DeletePersistentNotificationsPost", "app.user.get.app_error", nil, "", http.StatusInternalServerError).Wrap(nErr)
+			}
+		}
+		if user.IsGuest() {
+			c.Logger().Debug("DeletePersistentNotificationsPost: Persistent Notification feature is not enabled for guests.")
+			return nil
+		}
+	}
+
+	// if checkMentionedUser is "false" that would mean user is already authorized to delete the persistent-notification
+	// and the mention-validation can be skipped, so isUserMentioned will be "true".
+	isUserMentioned := !checkMentionedUser
+
+	// post owner is not allowed to stop the persistent notifications via ack, reply or reaction.
+	if checkMentionedUser && loggedInUserID != post.UserId {
+		if err := a.forEachPersistentNotificationPost([]*model.Post{post}, func(_ *model.Post, _ *model.Channel, _ *model.Team, mentions *ExplicitMentions, _ model.UserMap, _ map[string]map[string]model.StringMap) error {
+			if mentions.isUserMentioned(loggedInUserID) {
+				isUserMentioned = true
+			}
+			return nil
+		}); err != nil {
+			return model.NewAppError("DeletePersistentNotificationsPost", "app.post_priority.delete_persistent_notification_post.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+	}
+
+	if isUserMentioned {
+		if err := a.Srv().Store().PostPersistentNotification().Delete([]string{post.Id}); err != nil {
+			return model.NewAppError("DeletePersistentNotificationsPost", "app.post_priority.delete_persistent_notification_post.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+	}
+
+	return nil
+}
+
+func (a *App) SendPersistentNotifications() error {
+	notificationInterval := time.Duration(*a.Config().ServiceSettings.PersistentNotificationIntervalMinutes) * time.Minute
+	notificationMaxCount := int16(*a.Config().ServiceSettings.PersistentNotificationMaxCount)
+
+	// fetch posts for which first notificationInterval duration has passed
+	maxCreateAt := time.Now().Add(-notificationInterval).UnixMilli()
+
+	// Pagination loop
+	for {
+		notificationPosts, err := a.Srv().Store().PostPersistentNotification().Get(model.GetPersistentNotificationsPostsParams{
+			MaxCreateAt:   maxCreateAt,
+			MaxLastSentAt: maxCreateAt,
+			MaxSentCount:  notificationMaxCount,
+		})
```

Comment:
Might we simplify things by pushing the logic into the sql store? Seems unnecessary to pass in `maxCreateAt` twice instead of just letting the store return `GetPendingNotificationsPosts` directly.

Reply:
You mean have this logic directly into SQL?
```go
notificationInterval := time.Duration(*a.Config().ServiceSettings.PersistentNotificationIntervalMinutes) * time.Minute
	notificationMaxCount := int16(*a.Config().ServiceSettings.PersistentNotificationMaxCount)

	// fetch posts for which first notificationInterval duration has passed
	maxCreateAt := time.Now().Add(-notificationInterval).UnixMilli()
```
I'm not sure if we can rely on fetching the config values within a DB query. If those are not set, we would need the default values from the app.

Reply:
No, I'm wondering a simpler interface with the SQL store might look something like:
```go
notificationPosts, err := a.Srv().Store().PostPersistentNotification().Get(model.GetPersistentNotificationsPostsParams{
    since: time.Now().Add(-time.Duration(*a.Config().ServiceSettings.PersistentNotificationIntervalMinutes) * time.Minute),
    maxSentCount: *a.Config().ServiceSettings.PersistentNotificationMaxCount
})
```